neuralcompression/layers/_hyper_analysis_transformation_2d.py

```diff
@@ -14,26 +16,52 @@ class HyperAnalysisTransformation2D(Module):
     signal from a latent representation.
 
     The method is described in:
+
         | “Variational Image Compression with a Scale Hyperprior”
         | Johannes Ballé, David Minnen, Saurabh Singh, Sung Jin Hwang,
             Nick Johnston
         | https://arxiv.org/abs/1802.01436
 
     Args:
-        in_channels: number of channels in the input signal.
-        out_channels: number of channels produced by the transformation.
+        network_channels: number of channels in the input signal.
+        compression_channels: number of channels produced by the transformation.
     """
 
-    def __init__(self, in_channels: int, out_channels: int):
+    def __init__(
+        self,
+        network_channels: int,
+        compression_channels: int,
+        in_channels: int = 3,
+        activation: Module = ReLU(inplace=True),
+    ):
         super(HyperAnalysisTransformation2D, self).__init__()
 
-        self.models = Sequential(
-            Conv2d(in_channels, out_channels, (3, 3), (1, 2), 2),
-            ReLU(inplace=True),
-            Conv2d(out_channels, out_channels, (5, 5), (2, 2), 2),
-            ReLU(inplace=True),
-            Conv2d(out_channels, out_channels, (5, 5), (2, 2), 2),
+        self.encode = Sequential(
+            AbsoluteValue(),
+            Conv2d(
+                compression_channels,
+                network_channels,
+                (in_channels, in_channels),
+                (1, 1),
+                (in_channels // 2, in_channels // 2),
+            ),
+            activation,
```

Comment:
Looks like this uses the same object for each part of sequential, which has me a little worried about how this might work for saving activations for backward passes. Can we create a new class for each layer?

Reply:
Sure. Do you have any suggestions on doing that?

Reply:
If we narrow the scope to the scale hyperprior then we can just remove this option and use `ReLU()` everywhere.